src/goto-harness/goto_harness_parse_options.cpp

```diff
@@ -51,12 +51,35 @@ static void filter_goto_model(
   goto_modelt &goto_model_with_harness,
   const std::unordered_set<irep_idt> &goto_model_without_harness_symbols)
 {
-  for(auto &symbol : goto_model_with_harness.get_symbol_table())
+  for(auto const &symbol_id : goto_model_without_harness_symbols)
   {
-    if(
-      goto_model_without_harness_symbols.count(symbol.first) == 1 &&
-      symbol.second.is_function())
-      goto_model_with_harness.unload(symbol.first);
+    auto &symbol =
+      goto_model_with_harness.symbol_table.get_writeable_ref(symbol_id);
+    if(symbol.is_function())
+    {
+      // We don’t want bodies of functions that already existed in the
+      // symbol table (i.e. not generated by us)
+      goto_model_with_harness.unload(symbol_id);
+      if(symbol.is_file_local)
```

Comment:
Heads up this had to be changed because my stupid brain forgot that you can only take one branch in an if-else chain.